src/handlers/diff_header.rs

```diff
@@ -76,20 +67,48 @@ impl<'a> StateMachine<'a> {
                 &self.line,
                 &self.raw_line,
                 &mut self.painter,
+                &mut self.mode_info,
                 self.config,
             )?;
             handled_line = true;
         }
         Ok(handled_line)
     }
 
+    pub fn handle_diff_header_mode_line(&mut self) -> std::io::Result<bool> {
+        if !self.line.starts_with("old mode ") && !self.line.starts_with("new mode ") {
+            return Ok(false);
+        }
+        self.state = State::DiffHeader(DiffType::Unified);
+        if !self.should_handle() || self.config.color_only {
+            return Ok(false);
+        }
+        if self.line.starts_with("old") {
+            self.mode_info = self.line[9..].to_string();
```

Comment:
Can we make it clear where 9 comes from, either via the code or in a comment? Could this perhaps be rewritten as

```rust
if let Some(mode_info) = self.line.strip_prefix("old mode ")
```
?

(and below for `new_mode`)

Reply:
Looks like when I snagged that 9.. idiom from your code I failed to include your trailing comment about "old mode".len(). In any case, the strip_prefix() idiom is nicer.